megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -1724,18 +1733,31 @@ int getHighestEnemyInitiativeId() {
     private void performPathPostProcessing(final RankedPath path) {
         evadeIfNotFiring(path);
         unloadTransportedInfantry(path);
+        
+        // if we are using vector movement, there's a whole bunch of post-processing that happens to
+        // aircraft flight paths when a player does it, so we apply it here.
+        if(path.getPath().getEntity().isAero()) {
+            SharedUtility.moveAero(path.getPath(), null);
+        }
     }
     
     /**
      * Helper function that insinuates an "evade" step for aircraft that will not be shooting.
      * @param path The path to process
      */
     private void evadeIfNotFiring(final RankedPath path) {
+        Entity pathEntity = path.getPath().getEntity();
+        
+        // we cannot evade if we are out of control
+        if(pathEntity.isAirborne() && ((Aero) pathEntity).isOutControlTotal()) {
```

Comment:
Tracing back through the code I could not find anything that limits this line to aerospace units. `isAirborne()` returns true for non-aero units executing a combat drop, which will throw a ClassCastException here. In addition to a check for aero, it should also be cast to IAero so it can apply to LAMs in fighter mode.